common/internal/requestlogger.go

```diff
@@ -80,10 +82,11 @@ func (r *nopLogger) LogResponse(_ *http.Response)
 func (r *nopLogger) ResponseWriter(base http.ResponseWriter) http.ResponseWriter { return base }
 func (r *nopLogger) FlushLog()                                                   {}
 
-func NewRequestLogger(entry *logrus.Entry, req *http.Request) (RequestLogger, *logrus.Entry) {
-	if entry.Logger.IsLevelEnabled(logrus.DebugLevel) {
+func NewRequestLogger(ctx context.Context, req *http.Request, isDebug bool) (RequestLogger, context.Context) {
+	//nolint: // TODO: store debug logging flag against context (outside of logger): Context#withValue
```

Comment:
> Add the following function to support the `pkg` logger

Refer to the comment with the above text to help inform you about this area. Basically you want to store the verbose state of the logger in the context and pull it out here. The reason for this approach is that, while it's possible to use the `requestLogger` for each request and just allow the `pkg` logger to decide whether or not to log a particular entry, I'd prefer to avoid reading the body twice and caching it in a byte array if we have no intention of doing anything with it.